In [8]:
#from tensorflow_docs.vis import embed
from tensorflow import keras
#from imutils import paths

#import matplotlib.pyplot as plt
import tensorflow as tf
import pandas as pd
import numpy as np
#import imageio
import cv2
from ipywidgets import Video, Image

import os
import json

Building V2_trailer_Dataset

In [39]:
# # reading the JSON data using json.load()
# file = '/home/app/src/data/shot-type-dataset/v2_full_trailer.json'
# with open(file) as train_file:
#     dict_v2= json.load(train_file)

In [40]:
# Creating a DataFrame with the v2_full_trailer.json"

# dataset = []
# movie = []
# shot= []
# scale_label = []
# scale_val = []
# mov_label = []
# mov_val = []

# # We pass for all JSON keys an append every trailer labels different lists
# for key_movie in dict_v2:
#     for key_shot in dict_v2[key_movie]:

#         dataset.append('v2')
#         movie.append(key_movie)
#         shot.append(key_shot)
#         scale_label.append('None')
#         scale_val.append(int(0))
#         mov_label.append(dict_v2[key_movie][key_shot]['movement']['label'])
#         mov_val.append(dict_v2[key_movie][key_shot]['movement']['value'])

# # We build the dataframe with the lists data
# data = list(zip(dataset, movie, shot, scale_label, scale_val, mov_label, mov_val))  
# columns = ['dataset','movie', 'shot', 'scale_label', 'scale_val', 'move_label', 'move_val']  
        
# dataset_df_v2 = pd.DataFrame(data, columns=columns)

In [41]:
# v2_df = dataset_df_v2
# v2_path = '/home/app/src/data/shot-type-dataset/trailer_v2/'
# v2_df ['video_name'] = v2_path + dataset_df_v2['movie'] + '/'+ 'shot_' + dataset_df_v2['shot'] + '.mp4'
# v2_df = v2_df[['video_name', 'move_label']]
# v2_df = v2_df.rename(columns ={'move_label': 'tag'})


In [2]:
# v2_df.to_csv('/home/app/src/data/CSV/dataset_v2.csv', index = False)

Getting df of trailer_v2 movies

In [9]:
# Getting df of trailer_vw movies
v2_df= pd.read_csv('/home/app/src/CSV/dataset_v2.csv')
v2_df

video_name     tag
0    /home/app/src/data/shot-type-dataset/trailer_v...    Pull
1    /home/app/src/data/shot-type-dataset/trailer_v...    Push
2    /home/app/src/data/shot-type-dataset/trailer_v...    Push
3    /home/app/src/data/shot-type-dataset/trailer_v...    Push
4    /home/app/src/data/shot-type-dataset/trailer_v...    Pull
..                                                 ...     ...
605  /home/app/src/data/shot-type-dataset/trailer_v...    Push
606  /home/app/src/data/shot-type-dataset/trailer_v...  Static
607  /home/app/src/data/shot-type-dataset/trailer_v...  Static
608  /home/app/src/data/shot-type-dataset/trailer_v...  Motion
609  /home/app/src/data/shot-type-dataset/trailer_v...  Motion

[610 rows x 2 columns]

Preparing Trailer_v3_dataset (The complete dataset)

In [12]:
# data_df = pd. read_csv('/home/app/src/CSV/dataset_df.csv', dtype={"shot": str})

# data_df.drop(columns=['scale_val'], inplace=True)
# data_df.drop(columns = ['move_val'], inplace=True)

# train_path = '/home/app/src/data/shot-type-dataset/trailer_v3/train/'
# data_df['video_name'] = train_path + 'shot_' + data_df['movie'] + '_' + data_df['shot'] + '.mp4'
# train_df = data_df[(data_df['dataset']=='train') | (data_df['dataset']=='val') ]
# train_df = train_df[['video_name', 'move_label']]
# train_df = train_df.rename(columns ={'move_label': 'tag'})


# test_path = '/home/app/src/data/shot-type-dataset/trailer_v3/test/'
# data_df['video_name'] = test_path + 'shot_' + data_df['movie'] + '_' + data_df['shot'] + '.mp4'
# test_df = data_df[data_df['dataset']=='test']
# test_df = test_df[['video_name', 'move_label']]
# test_df = test_df.rename(columns ={'move_label': 'tag'})

Deleting Multi-movement Tag


In [13]:
# train_df = train_df[train_df.tag != 'Multi_movement']
# train_df = train_df.reset_index(drop = True) 
# test_df = test_df[test_df.tag != 'Multi_movement']
# test_df = test_df.reset_index(drop = True) 

In [14]:
# train_df.to_csv('/home/app/src/CSV/dataset_train_v3.csv', index = False)
# test_df.to_csv('/home/app/src/CSV/dataset_test_v3.csv', index = False)


Getting dfs (train and test) of trailer_v3 movies

In [20]:
train_df= pd.read_csv('/home/app/src/CSV/dataset_train_v3.csv')
test_df= pd.read_csv('/home/app/src/CSV/dataset_test_v3.csv')
train_df.tag.value_counts()

Static    16938
Motion     7386
Push        773
Pull        247
Name: tag, dtype: int64

Data Augmentation

We'll prepare a df with the videos to augmentate.

530 push videos from v2_df
 73 push videos from v2_df
247 push videos from v3_df

853 total videos which it'll augment into (853 x 11 types of aug) = 9383



In [28]:
#trailer_v2
v2_df_push_pull = v2_df[(v2_df['tag'] =='Pull') | (v2_df['tag'] =='Push')]
v2_df_push_pull

video_name   tag
0    /home/app/src/data/shot-type-dataset/trailer_v...  Pull
1    /home/app/src/data/shot-type-dataset/trailer_v...  Push
2    /home/app/src/data/shot-type-dataset/trailer_v...  Push
3    /home/app/src/data/shot-type-dataset/trailer_v...  Push
4    /home/app/src/data/shot-type-dataset/trailer_v...  Pull
..                                                 ...   ...
601  /home/app/src/data/shot-type-dataset/trailer_v...  Push
602  /home/app/src/data/shot-type-dataset/trailer_v...  Push
603  /home/app/src/data/shot-type-dataset/trailer_v...  Push
604  /home/app/src/data/shot-type-dataset/trailer_v...  Push
605  /home/app/src/data/shot-type-dataset/trailer_v...  Push

[606 rows x 2 columns]

In [31]:
#trailer_v3
v3_df_pull = train_df[(train_df['tag'] =='Pull')]
v3_df_pull

video_name   tag
8      /home/app/src/data/shot-type-dataset/trailer_v...  Pull
155    /home/app/src/data/shot-type-dataset/trailer_v...  Pull
208    /home/app/src/data/shot-type-dataset/trailer_v...  Pull
321    /home/app/src/data/shot-type-dataset/trailer_v...  Pull
447    /home/app/src/data/shot-type-dataset/trailer_v...  Pull
...                                                  ...   ...
24930  /home/app/src/data/shot-type-dataset/trailer_v...  Pull
24933  /home/app/src/data/shot-type-dataset/trailer_v...  Pull
25280  /home/app/src/data/shot-type-dataset/trailer_v...  Pull
25284  /home/app/src/data/shot-type-dataset/trailer_v...  Pull
25335  /home/app/src/data/shot-type-dataset/trailer_v...  Pull

[247 rows x 2 columns]

In [32]:
df_videos_to_aug= pd.concat([v2_df_push_pull,v3_df_pull], axis=0)
df_videos_to_aug=df_videos_to_aug.reset_index(drop=True)
df_videos_to_aug

video_name   tag
0    /home/app/src/data/shot-type-dataset/trailer_v...  Pull
1    /home/app/src/data/shot-type-dataset/trailer_v...  Push
2    /home/app/src/data/shot-type-dataset/trailer_v...  Push
3    /home/app/src/data/shot-type-dataset/trailer_v...  Push
4    /home/app/src/data/shot-type-dataset/trailer_v...  Pull
..                                                 ...   ...
848  /home/app/src/data/shot-type-dataset/trailer_v...  Pull
849  /home/app/src/data/shot-type-dataset/trailer_v...  Pull
850  /home/app/src/data/shot-type-dataset/trailer_v...  Pull
851  /home/app/src/data/shot-type-dataset/trailer_v...  Pull
852  /home/app/src/data/shot-type-dataset/trailer_v...  Pull

[853 rows x 2 columns]

In [33]:
#df_videos_to_aug.to_csv('/home/app/src/CSV/df_videos_to_aug.csv', index = False)